### PR TITLE
[TACHYON-1030] add unit test coverage for BlockMaster

### PR DESCRIPTION
**Pull request by aaudiber**
https://tachyon.atlassian.net/browse/TACHYON-1030

### CODE REVIEW

**Comment by haoyuan** [comment]
Jenkins, add to whitelist.

**Comment by haoyuan** [comment]
Jenkins, test this please.

File: servers/src/main/java/tachyon/master/block/BlockMaster.java [hunk 1]
```diff
@@ -607,6 +607,45 @@ private BlockInfo generateBlockInfo(MasterBlockInfo masterBlockInfo) {
   }
 
   /**
+   * This method should only be used by unit tests.
+   *
+   * @param worker a {@link MasterWorkerInfo} to add to the list of lost workers
+   */
+  void addLostWorker(MasterWorkerInfo worker) {
```
Comment (yupeng9) [inline servers/src/main/java/tachyon/master/block/BlockMaster.java]: Could you take a look at the class Tester and BlockDataManager.PrivateAccess to see how we use the private access pattern for adding methods only used by unit tests?

**Comment by yupeng9** [comment]
Thanks for the test improvement! We employ a private access pattern for hiding methods that are only for used testing purpose. Could you adopt that pattern in this change?

File: servers/src/test/java/tachyon/master/block/BlockMasterTest.java [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+ * Unit tests for tachyon.master.block.BlockMaster.
+ */
+public class BlockMasterTest {
+
+  @Rule
+  public TemporaryFolder mTestFolder = new TemporaryFolder();
+
+  @Test
+  public void countBytesTest() throws Exception {
+    BlockMaster master = createTestBlockMaster();
```
Comment (yupeng9) [inline servers/src/test/java/tachyon/master/block/BlockMasterTest.java]: We can make mater a class member variable, and construct it in before()

File: servers/src/main/java/tachyon/master/block/BlockMaster.java (hunk 1 above)
Comment (aaudiber) [inline servers/src/main/java/tachyon/master/block/BlockMaster.java]: Neat pattern! Even better than @VisibleForTesting! :)

Code has been updated

File: servers/src/test/java/tachyon/master/block/BlockMasterTest.java (hunk 2 above)
Comment (aaudiber) [inline servers/src/test/java/tachyon/master/block/BlockMasterTest.java]: Done

**Comment by yupeng9** [comment]
LGTM

**Comment by yupeng9** [comment]
Thanks for the improvement! Also, going forward, please keep the git history (e.g. avoid commit --amend after the push), so that reviewers can track the history of comments.

**Pull request merged**